src/main/java/net/earthcomputer/clientcommands/features/TwoPlayerGame.java

```diff
@@ -0,0 +1,182 @@
+        }
+
+        StartTwoPlayerGameC2CPacket packet = new StartTwoPlayerGameC2CPacket(source.getClient().getConnection().getLocalGameProfile().getName(), false, this);
+        C2CPacketHandler.getInstance().sendPacket(packet, recipient);
+        this.pendingInvites.add(recipient.getProfile().getName());
+        source.sendFeedback(Component.translatable("c2cpacket.startTwoPlayerGameC2CPacket.outgoing.invited", recipient.getProfile().getName(), translate()));
+        return Command.SINGLE_SUCCESS;
+    }
+
+    public int open(FabricClientCommandSource source, String name) throws CommandSyntaxException {
+        T game = this.activeGames.get(name);
+        if (game == null) {
+            throw NO_GAME_WITH_PLAYER_EXCEPTION.create();
+        }
+
+        source.getClient().schedule(() -> source.getClient().setScreen(this.screenFactory.createScreen(game)));
+        return Command.SINGLE_SUCCESS;
+    }
+
+    public static void onStartTwoPlayerGame(StartTwoPlayerGameC2CPacket packet) {
+        String sender = packet.sender();
+        TwoPlayerGame<?, ?> game = packet.game();
+        PlayerInfo opponent = Minecraft.getInstance().getConnection().getPlayerInfo(sender);
+        if (opponent == null) {
+            return;
+        }
+
+        if (packet.accept() && game.getPendingInvites().remove(sender)) {
+            packet.game().addNewGame(opponent, true);
+
+            MutableComponent component = Component.translatable("c2cpacket.startTwoPlayerGameC2CPacket.incoming.accepted", sender, game.translate());
+            component.withStyle(style -> style
+                .withClickEvent(new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/" + game.command + " open " + sender))
+                .withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, Component.literal("/" + game.command + " open " + sender))));
+            Minecraft.getInstance().gui.getChat().addMessage(component);
+            return;
+        }
+
+        MutableComponent component = Component.translatable("c2cpacket.startTwoPlayerGameC2CPacket.incoming", sender, game.translate());
+        component
+            .append(" [")
+            .append(Component.translatable("c2cpacket.startTwoPlayerGameC2CPacket.incoming.accept").withStyle(style -> style
+                .withColor(ChatFormatting.GREEN)
+                .withClickEvent(new ClickEvent(ClickEvent.Action.CHANGE_PAGE, ClientCommandHelper.registerCode(() -> {
+                    game.addNewGame(opponent, false);
+
+                    StartTwoPlayerGameC2CPacket acceptPacket = new StartTwoPlayerGameC2CPacket(Minecraft.getInstance().getConnection().getLocalGameProfile().getName(), true, game);
+                    try {
+                        C2CPacketHandler.getInstance().sendPacket(acceptPacket, opponent);
+                    } catch (CommandSyntaxException e) {
+                        Minecraft.getInstance().gui.getChat().addMessage(Component.translationArg(e.getRawMessage()));
```

Comment:
I see you have this in a few places. We have a utility `ClientCommandHelper.sendFeedback` for it which is much shorter